src/invidious/videos/caption.cr

```diff
@@ -31,6 +31,72 @@ module Invidious::Videos
       return captions_list
     end
 
+    def timedtext_to_vtt(timedtext : String, tlang = nil) : String
+      # In the future, we could just directly work with the url. This is more of a POC
+      cues = [] of XML::Node
+      tree = XML.parse(timedtext)
+      tree = tree.children.first
+
+      tree.children.each do |item|
+        if item.name == "body"
+          item.children.each do |cue|
+            if cue.name == "p" && !(cue.children.size == 1 && cue.children[0].content == "\n")
+              cues << cue
+            end
+          end
+          break
+        end
+      end
+      result = String.build do |result|
+        result << <<-END_VTT
+        WEBVTT
+        Kind: captions
+        Language: #{tlang || @language_code}
+
+
+        END_VTT
+
+        result << "\n\n"
```

Comment:
And this one is only needed if you don't keep the two empty lines before `END_VTT`!